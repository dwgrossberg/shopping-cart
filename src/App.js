import React, { useState, useEffect } from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Header from "./components/Header/Header";
import Footer from "./components/Footer/Footer";
import Home from "./pages/home/home";
import Products from "./pages/products/products";
import Contact from "./pages/contact/contact";
import "./styles/App.scss";
import ScrollToTop from "./components/ScrollToTop";

const App = () => {
  const [products, setProducts] = useState([]);
  const [cart, setCart] = useState([]);
  const [category, setCategory] = useState([]);

  // shuffle array to randomize product display on each visit
  const shuffle = (array) => {
    let currentIndex = array.length;
    let randomIndex;
    while (currentIndex !== 0) {
      randomIndex = Math.floor(Math.random() * currentIndex);
      currentIndex--;
      [array[currentIndex], array[randomIndex]] = [
        array[randomIndex],
        array[currentIndex],
      ];
    }
    return array;
  };

  // fetch product data from fake store api
  const fetchProductData = async () => {
    try {
      const prodResponse = await fetch("https://fakestoreapi.com/products");
      const prodData = await prodResponse.json();
      const shuffledData = shuffle(prodData);
      setProducts(shuffledData);
      setCategory(shuffledData);
      console.log(prodData);
      return prodData;
    } catch (err) {
      console.log(err);
    }
  };

  // fetch cart data from fake store api
  const fetchCartData = async () => {
    const randomNum = Math.floor(Math.random() * 6 + 1);
    try {
      const cartResponse = await fetch(
        `https://fakestoreapi.com/carts/${randomNum}`
      );
      const cartData = await cartResponse.json();
      setCart(cartData.products);
      console.log(cartData.products);
      return cartData;
    } catch (err) {
      console.log(err);
    }
  };

  useEffect(() => {
    fetchProductData();
    fetchCartData();
  }, []);

  return (
    <div className="App">
      <BrowserRouter>
        <Header cart={cart} />
        <ScrollToTop />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route
            path="/products"
            element={
              <Products
                products={products}
                setProducts={setProducts}
                category={category}
                setCategory={setCategory}
                cart={cart}
                setCart={setCart}
                shuffle={shuffle}
              />
            }
          />
          <Route path="/contact" element={<Contact />} />
        </Routes>
        <Footer />
      </BrowserRouter>
    </div>
  );
};

export default App;
