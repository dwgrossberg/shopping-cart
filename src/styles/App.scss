.App {
  height: 100%;
  min-height: (100vh);
  width: 100%;
  gap: 20px;
  display: grid;
  align-items: stretch;
  justify-items: center;
  grid-template: 1fr 50px / 1fr;
  padding: 30px;
  background-color: ghostwhite;
}

.price-quant {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 80%;
  gap: 20px;
  .quant-number {
    width: 10px;
    text-align: center;
    font-size: 9px;
  }
  .quant {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 10px;
    button {
      width: 18px;
      height: 18px;
      border-radius: 7px;
      background-color: white;
      border: 0.5px solid #af9ab2;
      cursor: pointer;
      &:hover {
        background-color: #672a4e;
        color: white;
      }
    }
  }
}
