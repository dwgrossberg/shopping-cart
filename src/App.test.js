import "@testing-library/jest-dom";
import React from "react";
import { screen, render } from "@testing-library/react";
import App from "./App";

describe("App component", () => {
  Object.defineProperty(window, "scrollTo", {
    value: () => {},
    writable: true,
  });
  it("snapshot test", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("renders correct heading", () => {
    render(<App />);
    expect(screen.getByRole("heading").textContent).toMatch(
      /anything you want store/i
    );
  });
});
