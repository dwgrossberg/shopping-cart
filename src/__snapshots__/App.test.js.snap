// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <nav
        class="header-nav"
      >
        <ul>
          <li>
            <img
              alt="icon"
              id="icon"
              src="icon.png"
            />
          </li>
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
            style="text-decoration: none;"
          >
            <li>
              home
            </li>
          </a>
          <a
            class="nav-link"
            href="/products"
            style="text-decoration: none;"
          >
            <li>
              products
            </li>
          </a>
          <a
            class="nav-link"
            href="/contact"
            style="text-decoration: none;"
          >
            <li>
              contact
            </li>
          </a>
        </ul>
      </nav>
      <a
        class=""
        href="/cart"
      >
        <div
          id="cart"
        >
          <img
            alt="cart"
            src="cart.png"
          />
        </div>
      </a>
    </div>
    <div
      class="home"
    >
      <h2>
        anything you want store
      </h2>
    </div>
    <div
      class="footer"
    >
      <div
        class="made-by"
      >
        <p>
          Made by
        </p>
        <a
          href="https://github.com/dwgrossberg"
        >
          <img
            alt="github-icon"
            id="github-icon"
            src="github.png"
          />
        </a>
        <p>
          Dan Grossberg
        </p>
      </div>
    </div>
  </div>
</div>
`;
